src/main/scripts/jenkins

```diff
@@ -2,33 +2,66 @@
 
 set -e
 
-nprocs="$(grep -c '^processor' /proc/cpuinfo)"
+notif() {
+    echo -e "\n$@" \
+            "\n================================================================================" \
+            "\n" >&2
+}
 
+NPROCS="${NPROCS:-$(grep -c '^processor' /proc/cpuinfo)}"
+WORKSPACE="${WORKSPACE:-$(pwd)}"
+
+K_EXERCISES_REPO="${K_EXERCISES_REPO:-git@github.com:kframework/k-exercises.git}"
+RVMATCH_REPO="${RVMATCH_REPO:-git@github.com:runtimeverification/rv-match.git}"
+
+KSERVER_SOCKET="$WORKSPACE/kserver"
+KSERVER_LOG="$WORKSPACE/kserver.log"
+export KSERVER_SOCKET
+
+notif "CLEANING REPO"
+mvn clean
+git clean -dffx
+
+notif "SETTING UP OPAM/PERL ENVIRONMENTS"
 eval `opam config env`
 eval `perl -I$HOME/perl5/lib/perl5 -Mlocal::lib`
 
-echo "=== RUNNING LOCAL TESTS ==="
-cd ${WORKSPACE}/k-pr
+notif "CLONING SUBMODULES"
 git submodule update --init
+
+notif "BUILDING/RUNNING LOCAL TESTS"
 mvn verify -U
 
-echo "=== RUNNING K EXERCISES ==="
-cd ${WORKSPACE}/k-pr
-rm -rf k-exercises
-git clone 'git@github.com:kframework/k-exercises.git'
-cd k-exercises/tutorial
-export KSERVER_SOCKET=$(pwd)/kserver
-../../k-distribution/target/release/k/bin/spawn-kserver kserver.log
+notif "STARTING KSERVER"
+./k-distribution/target/release/k/bin/spawn-kserver "$KSERVER_LOG"
 sleep 5
-make -j"$nprocs"
-../../k-distribution/target/release/k/bin/stop-kserver || true
 
-echo "=== RUNNING RV-MATCH TESTS ==="
-cd ${WORKSPACE}
+notif "CLONING K EXERCISES"
+rm -rf k-exercises
+git clone "$K_EXERCISES_REPO" k-exercises
+
+notif "RUNNING K EXERCISES"
+exit_status='0'
+(   cd k-exercises/tutorial
+    make -j"$NPROCS"
+) || exit_status="$?"
```

Comment:
Has this been tested in the case it fails? I ask because the last time you wrote something in approximately this syntax I had to change it because it didn't actually work.

Reply:
```
[dev@arch-ehildenb k-pr]$ exit_status='0'
[dev@arch-ehildenb k-pr]$ echo "$exit_status"
0
[dev@arch-ehildenb k-pr]$ ( cd k-distribution/
> false
> ) || exit_status="$?"
[dev@arch-ehildenb k-pr]$ echo $exit_status
1
[dev@arch-ehildenb k-pr]$ 
```


The parenthesis spawn a sub-shell (so that any `cd` that happen don't get carried over past that part of the script). The semantics of `$?` is that the last command run (in sub-shell or otherwise) by the bash interpreter will be stored in `$?`. So if `cd ...` fails, then we'll get the `exit_status` set to the failure code of `cd`, and if `cd` succeeds and `make` fails, then `exit_status` will be whatever `make` returns.